### ISSUE TITLE
[Request] Stdin or pipe

### ISSUE THREAD

**Issue by rNoz** [ssato/python-anytemplate#4]
Hi,

It will be great that your library could accept the config `-C` from stdin or pipe, to "pass" config variables without writting them to a file first.

```sh
echo '{a: true, b: "something" }' | anytemplate_cli template.j2

echo 'a: true\nb: "something"\n' | anytemplate_cli template.j2

anytemplate_cli template.j2 < EOF
---
a: true
b: "something"
EOF
```

Can you provide this functionality?

**Comment by ssato** [comment]
Excuse me to late reply.

I took some to think and decided not to implement this that because I think there is no consistent way to do that, sorry.

Templates are the inputs for anytemplate_cli and context (config) files are just aux data originally. I feel it's not consistent if anytemplate_cli reads some data from stdin or pipe and processes it as context (config) data instead of templates.

**Comment by ssato** [comment]
Last year, I though that this is very hard to implement consistently but I revisited and think I found a solution.
Several commits like 064bfa1 and 6e32375 add support to load template and context (config) content from stdin.

Here are some examples:
```
ssato@localhost% cat /tmp/ctx.yml
a: aaa
ssato@localhost% echo '$a' | PYTHONPATH=. python anytemplate/cli.py -E string.Template -C /tmp/ctx.yml -o - -
aaa

ssato@localhost% cat /tmp/test.tmpl
$a
ssato@localhost% echo 'a: aaa' | PYTHONPATH=. python anytemplate/cli.py -E string.Template -C yaml:- -o - /tmp/test.tmpl
aaa

ssato@localhost%
```

Please note that we have to give context (config) type ("yaml" in -C yaml:- option) if we want to make it read from stdin because there is no way to detect its type automatically in this case.

**Comment by ssato** [comment]
I've released 0.1.4 (https://pypi.org/project/anytemplate/) contains enhancements for this and close this. If you have any further issues related to this, please open this again or another issue.
